core/src/main/java/be/orbinson/aem/dictionarytranslator/services/DictionaryService.java

```diff
@@ -1,192 +1,131 @@
      */
-    public final class Message {
-        private final String text;
-        private final String resourcePath;
-
-        public Message(String message, String resourcePath) {
-            this.text = message;
-            this.resourcePath = resourcePath;
-        }
-
-        public String getText() {
-            return text;
-        }
-
-        public Optional<String> getResourcePath() {
-            return Optional.ofNullable(resourcePath);
-        }
-
-        @Override
-        public String toString() {
-            return "Message [text=" + text + ", resourcePath=" + resourcePath + "]";
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(text, resourcePath);
-        }
-
-        @Override
-        public boolean equals(Object obj) {
-            if (this == obj)
-                return true;
-            if (obj == null)
-                return false;
-            if (getClass() != obj.getClass())
-                return false;
-            Message other = (Message) obj;
-            return Objects.equals(text, other.text) && Objects.equals(resourcePath, other.resourcePath);
-        }
-    }
-
-    enum DictionaryType {
-        SLING_MESSAGE_ENTRY("sling:MessageEntry"),
-        JSON_FILE("JSON file"),
-        /** the languages below this dictionary have different types, only applicable on dictionary level, not on language level */
-        MIXED("Mixed");
-
-        private final String label;
-
-        DictionaryType(String label) {
-            this.label = label;
-        }
-
-        @Override
-        public String toString() {
-            return label;
-        }
-    }
+    void createDictionaries(ResourceResolver resourceResolver, String parentPath, String name, Collection<Locale> languages, Collection<String> basenames) throws PersistenceException, DictionaryException;
 
     /**
-     * Returns the type of the dictionary for the given language.
-     * @param dictionaryResource
-     * @param language
-     * @return the type of the dictionary for the given language
-     * @throws DictionaryException if the dictionary for the given language does not exist
+     * Deletes resource in the given container path containing all its children. This affects both dictionaries as well as other child resources.
+     * The changes are not persisted, so you need to call {@link ResourceResolver#commit()} afterwards.
+     * @param replicator
+     * @param resourceResolver
+     * @param containerPath
+     * @throws DictionaryException in case the container path does not exist
+     * @throws ReplicationException
+     * @throws PersistenceException
      */
-    DictionaryType getType(Resource dictionaryResource, String language) throws DictionaryException;
+    void deleteDictionaries(Replicator replicator, ResourceResolver resourceResolver, String containerPath) throws DictionaryException, ReplicationException, PersistenceException;
```

Comment:
whats the reasoning of not injecting the replicator directly in the dictionaryservice?

Reply:
This is not only used here but also used in methods of the LanguageDictionary (which is not an OSGi service). That is the reason why I haven't injected it any longer. Also for read access this is not necessary at all, therefore I have a slight preference to just passing it along when calling relevant methods.